### ISSUE TITLE
Want some translations ?

### ISSUE THREAD

**Issue by Piplopp** [arghblargh/feh-inheritance-tool#1]
Yo, I don't know how to JS but I can provide (at least) french translation for the skills if you want :)

I was thinking about adding some jsons files for the different languages.

**Comment by Piplopp** [comment]
Well, after taking a look at how it works, if you want some translations I think it may be good to tweak the jsons a bit.

I suggest adding an id field for each skill and tweaking a bit the [units.json](https://github.com/arghblargh/feh-inheritance-tool/blob/master/src/data/units.json) file to use the ids and not the skills names.

**Comment by arghblargh** [comment]
Sure! It'll be great to have them when I add support for other languages.

**Comment by arghblargh** [comment]
I was just thinking of using a name mapping instead of ids, but I can change to ids if necessary.

**Comment by Piplopp** [comment]
For god's sake, there is like 3 characters which have a different name in French ! I didn't expect that...
Due to the units changing names in different languages too I guess some translation of `units.json` is also mandatory.

- Switching to id mapping may prevent to duplicate some translations, A weapon id will be the same in all languages so the `units.json` file only needs this to avoid re-translating the same thing twice.

- However, due to a few (2-3 / language) of heroes with different names a (partial) translation of this file is needed too. I feel like a copy of the entire file for just 2-3 name changes by language is just unnecessary duplication.

I didn't find any good way to do this yet, only the ugly use of a default and language specific name:

```json
{
    "charID": {
        "name" : "Cherche",
        "nameFR" : "Zelcher",
        ...
    },
    ...
}
```
or (mostly the same)

```json
{
    "Cherche": {
        "nameFR" : "Zelcher",
        ...
    },
    ...
}
```
And when the _nameLANG_ doesn't exists, use the default one.

**TL;DR;** Switching to id mapping is easier and faster for translation purpose  due to `units.json`. A few heroes have their names changed too, need to find a good way to do this (I feel like a copy of the entire file for just 3 heroes per language is a waste).

**Comment by arghblargh** [comment]
I was thinking more along the lines of having a single file for each language with everything. Then I could look up the translations using the English name as a key:

```json
{
    "Cherche": {
      "name" : "セルジュ"
    },
    "Fortify Def 3": {
        "name" : "守備の鼓舞3",
        "effect" : "ターン開始時、周囲1マスの味方の守備＋4（1ターン）"
    },
    ...
}
```

or even just a single gigantic file:

```json
{
    "Cherche": {
        "name" :  {
            "JP" : "セルジュ"
            ...
        }
    },
    "Fortify Def 3": {
        "name" : {
            "JP" : "守備の鼓舞3"
            ...
        },
        "effect" : {
            "JP" : "ターン開始時、周囲1マスの味方の守備＋4（1ターン）"
            ...
        }
    },
    ...
}
```

**Comment by Piplopp** [comment]
The gigantic file will be a mess in the future and hardly maintainable I think.

The one file by language is barely the same as what you have now It's just a matter of organization preferences. I see no problem using the English names as an id, It actually a better idea.

I am not sure to understand what you meant on your 1st example. I understood it as adding a new json linking English names to other language names.
If that's so it's a really good idea, because the `units.json` wont have to be translated anymore and will do the job of linking skills to heroes. In all languages.

We just need a new json for units names. It can even be the simplest json ever (for example in the Japanese names translation):

_names_jap.json_
```json
{
    "Cherche": "セルジュ",
    ...
}
```
This will lead us to have either 5 jsons per language (assists, weapons, specials, passives, names) or one big with every thing in a particular language (just a matter of preference). Plus the units.json wich is defining what a hero is composed of.

If that's what you meant, I think this is a great idea.

**Comment by arghblargh** [comment]
Yeah, that's what I meant, with all data for each language in its own file, instead of having 5 files for each.

**Comment by Piplopp** [comment]
Great! One file is fine to me :)
I am closing the issue since it's all agreed :+1:

**Comment by Piplopp** [comment]
Reopening this for a quick question now that I managed to find some time.

For a complete `"one_file_per_lang"` you prefer something like this:

```json
{
    "Cherche": {
      "name" : "セルジュ"
    },
    "Fortify Def 3": {
        "name" : "守備の鼓舞3",
        "effect" : "ターン開始時、周囲1マスの味方の守備＋4（1ターン）"
    },
    ...
}
```

or with one more level to add a bit of structure ? It may also prevent finding a hero in the middle of the passives...
Like:


```json
{
    "HEROES": {
        "Cherche": {
          "name" : "セルジュ"
        },
        ...
    },
    "PASSIVE_A": {
        "Fortify Def 3": {
            "name" : "守備の鼓舞3",
            "effect" : "ターン開始時、周囲1マスの味方の守備＋4（1ターン）"
        },
        ...
    },
    "SPECIALS":{
        ...
    },
    ...
}
```

EDIT: removed something after thinking about it.

**Comment by arghblargh** [comment]
I think I'd prefer the first one. I doubt there will be things with the same name, but if it happens we can change it then.

**Comment by Piplopp** [comment]
Okay :)
